File: java/RemoveDuplicatesFromSortedArray.java
/*
 * Filename: RemoveDuplicatesFromSortedArray.java
 * Author: Jun Xu
 * Email: junx@bu.edu
 * Created Time: Mon 02 Mar 2015 07:48:41 PM EST
 */

public class Solution {
	public int removeDuplicates(int[] A) {
		int length = A.length;
		if (length == 0) return 0;
		else if (length == 1) return 1;
		else {
			int prev = 0;
			int curr = 1;
			int res = 1;

			for (; curr < length; curr++) {
				if (A[prev] != A[curr])	{
					res++;
					A[++prev] = A[curr];
				}
			}

			return res;
		}
	}
}

File: cpp/remove_duplicates_from_sorted_array.cc
/*
 * Filename: remove_duplicates_from_sorted_array.cc
 * Author: Jun Xu
 * Email: jxu.vimacs@gmail.com
 * Created Time: Fri 24 Jul 2015 02:21:06 PM EDT
 */

#include <iostream>
#include <vector>

using namespace std;

class Solution {
	public:
		int removeDuplicates(vector<int>& nums) {
			if (nums.size() < 2) {
				return nums.size();
			}

			int res = 0;
			for (vector<int>::size_type i = 0; i < nums.size(); i++) {
				if (i && nums[i] == nums[i-1]) {
					continue;
				}

				nums[res++] = nums[i];
			}

			return res;
		}
};
